core/domain/classifier_services.py

```diff
@@ -127,6 +127,52 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def create_classifier_training_jobs(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
+    exp_version = exploration.version
+    states = exploration.states
+    for state_name in state_names:
+        state = states[state_name]
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        create_classifier_training_job(algorithm_id, interaction_id,
+                                       exp_id, exp_version, state_name,
+                                       training_data,
+                                       feconf.TRAINING_JOB_STATUS_NEW)
+
+
+def update_classifier_exploration_mappings(exploration, state_names):
+    """Creates new ClassifierExplorationMappingModel instances for all the
+    state names passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
```

Comment:
@seanlip Ah. Regarding the mapping, there will never be a mapping for the v1 - mappings are normally created only when classifiers are stored. So, here if v1 and is not picked up by the VM, its status is 'NEW'. When v2 is committed and we retrieve v1's job, if the status is NEW, we delete v1's job and create a new job for v2. So, when the classifier for v2 is stored, the mapping for this will be created.

Oh. If the previous job's status if FAILED and not picked up again by the VM yet, we can delete that job as well.

If the previous job's status is PENDING, we can't help but let it get stored and its mapping be created. But, when v2's classifier is stored, v1's mapping becomes redundant because there's now a new mapping for v2.

Phew, that sounds confusing. There's a lot more stuff I need to add to this PR. So many corner cases we shouldn't miss.